.github/workflows/sanity-test.yaml

```
@@ -0,0 +1,244 @@
+name: Sanity Testing
+on: [push]
+
+permissions:
+  # required to retrieve AWS credentials
+  id-token: write
+  contents: write
+
+# cancel currently running jobs if a new version of the branch is pushed
+concurrency:
+  group: sanity_testing-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+
+jobs:
+  Sanity-Tests:
+    environment: Testing
+    runs-on: ubuntu-latest
+    env:
+      AZURE_CLIENT_ID: ${{ secrets.CLIENT_ID }}
+      AZURE_CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
+      AZURE_TENANT_ID: ${{ secrets.TENANT_ID }}
+      CORSO_M365_TEST_USER_ID: ${{ secrets.CORSO_M365_TEST_USER_ID }}
+      CORSO_PASSPHRASE: ${{ secrets.INTEGRATION_TEST_CORSO_PASSPHRASE }}
+      TEST_RESULT: "test_results"
+      CORSO_BUCKET: ${{ secrets.CI_TESTS_S3_BUCKET }}
+
+    defaults:
+      run:
+        working-directory: src
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Golang with cache
+        uses: magnetikonline/action-golang-cache@v3
+        with:
+          go-version-file: src/go.mod
+
+      - run: make build
+
+      - run: go build -o sanityCheck ./cmd/sanity_test
+
+      - run: mkdir test_results
+
+      # AWS creds
+      - name: Configure AWS credentials from Test account
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE }}
+          role-session-name: integration-testing
+          aws-region: us-east-1
+
+      # run the tests
+      - name: Version Test
+        run: |
+          set -euo pipefail
+          if [ $( ./corso --version | grep 'Corso version:' | wc -l) -ne 1 ]
+          then
+              echo "valid version not found"
+              exit 1
+          fi
+
+      - name: Repo init test
+        env:
+          TEST_RESULT: "test_results"
+        run: |
+          set -euo pipefail
+          touch $TEST_RESULT/initrepo.txt
+          prefix=`date +"%Y-%m-%d-%T"`
+          ./corso repo init s3 --hide-progress --prefix $prefix --bucket   ${CORSO_BUCKET}  2>&1 | tee $TEST_RESULT/initrepo.txt 
+          valid_num=$(grep 'Initialized a S3 repository within bucket' $TEST_RESULT/initrepo.txt | wc -l)
+
+          if [ $valid_num -ne 1 ]
+          then
+              echo "repo could not be initiated"
+              exit 1
+          fi
```

This check can be limited to just a grep. Same comment in a few other similar usages.
```suggestion
          if grep -q 'Initialized a S3 repository within bucket' $TEST_RESULT/initrepo.txt
          then
              echo "repo could not be initiated"
              exit 1
          fi
```

---

Also, wondering if we should just check the last line(`tail -n1`) instead of the whole output.